ui-manchette/src/components/Waypoint.tsx

```diff
@@ -6,42 +6,57 @@ import { type InteractiveWaypoint } from '../types';
 import '@osrd-project/ui-core/dist/theme.css';
 import { positionMmToKm } from '../utils';
 
+const MENU_OFFSET = 2;
+
 type WaypointProps = {
   waypoint: InteractiveWaypoint;
   isActive: boolean;
+  waypointMenu?: React.ReactNode;
 };
 
 const Waypoint = ({
   waypoint: { name, secondaryCode, id, position, display, onClick },
   isActive,
+  waypointMenu,
 }: WaypointProps) => {
-  const opRef = useRef<HTMLDivElement>(null);
+  const waypointRef = useRef<HTMLDivElement>(null);
 
   if (!display) return null;
 
   return (
-    <div
-      className={cx('flex waypoint items-baseline', {
-        'menu-active': isActive,
-      })}
-      id={id}
-      ref={opRef}
-      onClick={() => {
-        if (onClick) onClick(id, opRef.current);
-      }}
-    >
-      <div className="waypoint-position justify-self-start text-end">
-        {positionMmToKm(position)}
+    <>
+      <div
+        ref={waypointRef}
+        className={cx('flex waypoint items-baseline', {
+          'menu-active': isActive,
+        })}
+        id={id}
+        onClick={() => {
+          if (onClick) onClick(id);
+        }}
+      >
+        <div className="waypoint-position justify-self-start text-end">
+          {positionMmToKm(position)}
+        </div>
+
+        <div className="waypoint-name mx-2 justify-self-start">{name}</div>
+        <div className="waypoint-separator"></div>
+        <div className="waypoint-ch font-mono justify-self-end">{secondaryCode}</div>
+        <div className="waypoint-separator"></div>
+
+        <div className="waypoint-type"></div>
+        <div className="waypoint-separator"></div>
+        {waypointMenu && isActive && waypointRef.current && (
+          <div
+            className="menu-wrapper"
+            style={{ top: waypointRef.current.getBoundingClientRect().bottom - MENU_OFFSET }}
```

Comment:
This sets a position relative to the viewport, but this breaks as soon as a parent is set as `position: relative`. Since the manchette is a re-usable component, it shouldn't assume anything about the positioning mode of its parent.

I believe this won't work in OSRD since `.chart-container` has `position: relative`.